hphp/runtime/ext/datetime/ext_datetime.php

```diff
@@ -876,8 +875,21 @@ function timezone_name_get(DateTimeZone $timezone): string {
 }
 
 <<__ParamCoerceModeFalse>>
-function timezone_open(string $timezone): DateTimeZone {
-  return new DateTimeZone($timezone);
+function timezone_open(string $timezone): mixed {
+  try {
+    return new DateTimeZone($timezone);
+  }
+  catch (Exception $e) {
+    trigger_error(
+      str_replace(
+        'DateTimeZone::__construct():',
+        __FUNCTION__ . '():',
+        $e->getMessage()
+      ),
+      E_WARNING
```

Comment:
Yep, builtins are allowed to use any error constant, even if they're written in PHP or Hack: https://github.com/facebook/hhvm/blob/master/hphp/runtime/ext/std/ext_std_errorfunc.cpp#L291-L312

We needed this to maintain PHP-compatibility, while still allowing parts of the standard library to be implemented in PHP or Hack.